client/lib/signup/flow-controller.ts

```diff
@@ -151,6 +158,42 @@ export default class SignupFlowController {
 		}
 	}
 
+	_resetSiteSlugIfUserEnteredAnotherFlow() {
```

Comment:
`_resetStoresIfProcessing()` & `_resetStoresIfUserHasLoggedIn()` both reset the complete store, so why don't we do the same when `previousFlowName !== currentFlowName`? 

In this way we don't have to:
1. check for `siteSlug` in all steps dependencies
2. create new action types for removing the site slug
3. update the reducer for removing the site slug

Reply:
> `_resetStoresIfProcessing()` & `_resetStoresIfUserHasLoggedIn()` both reset the complete store, so why don't we do the same when `previousFlowName !== currentFlowName`?

Actually, you have a point.

I think it started from this comment https://github.com/Automattic/wp-calypso/pull/51021#issuecomment-809079985 where resetting the store would prevent user from resuming from where they left off if they return to the signup flow again (maybe refresh, maybe open in a new tab, etc.).

From there I was thinking, why not just remove the `siteSlug` instead of resetting the whole store? But in order to remove the `siteSlug`, you would have to determine whether the user is switching from one flow to another, and that logic had to be implemented.

But now, with that logic implemented and we have the ability to determine whether flow has changed, it kinda start to make a lot of sense to just reset the whole store when flows change, because, why carry over data from the previous store if the user is changing flows.

Unless there is a reason to retain data when transitioning from one flow to another, there are a lot of flows, I can't tell.

This needs a bit of flow investigation. If we decide to reset store when changing flows, we need to make sure that all flows are cool with that.

If not, removing siteSlug is now a less risky alternative because we know we definitely don't want to carry over this value from previous flow if there is a step in the new flow that provides this dependency.

Albeit, yes, removing siteSlug when changing flows does feel a bit like monkey patching, whereas resetting store when changing flows feels more like a proper fix (if all flows are cool with it).

Give me some time to think how to balance this out with https://github.com/Automattic/wp-calypso/issues/52610 because they are touching similar codes.

Reply:
Thank you so much for elaborating 👍 

> [...]where resetting the store would prevent user from resuming from where they left off[...]

> But now, with that logic implemented and we have the ability to determine whether flow has changed[...]

Exactly! Before we were able to determine if the flow has changed, resetting the store is potentially worsening the user experience since they might be unable to continue where they left off.  

> [...]we need to make sure that all flows are cool with that.

Absolutely!

> Give me some time to think how to balance this out with #52610 because they are touching similar codes.

Sure! Thank you so much for giving this a thought. I completely understand that resetting the store is risky and that you need more time to balance it out.